conduit/pipeline/pipeline.go

```diff
@@ -254,8 +274,26 @@ func (p *pipelineImpl) Init() error {
 
 	// InitProvider
 	round := sdk.Round(p.pipelineMetadata.NextRound)
-	// Initial genesis object is nil--gets updated after importer.Init
-	var initProvider data.InitProvider = conduit.MakePipelineInitProvider(&round, nil)
+
+	// Initialize Telemetry
+	var telemetryClient *telemetry.OpenSearchClient
+	if p.cfg.Telemetry.Enabled {
+		// If telemetry cannot be initialized, log a warning and continue
+		// pipeline initialization.
+		var telemetryErr error
+		telemetryClient, telemetryErr = p.initializeTelemetry()
+		if telemetryErr != nil {
+			p.logger.Warn("Telemetry initialization failed. Continuing without telemetry.")
+		}
+		// Try sending a startup event. If it fails, log a warning and continue
+		event := telemetryClient.MakeTelemetryStartupEvent()
+		if telemetryErr = telemetryClient.SendEvent(event); telemetryErr != nil {
+			p.logger.Warnf("failed to send telemetry event: %s", telemetryErr)
+		}
```

Comment:
What do you think about wrapping the OpenSearch client in our own object that implements SendEvent?

It could manage the logging and allow us to swap in a different client in the future if it became useful. Since this is part of the public interface, it could be problematic to swap out or upgrade the client in the future.